### PR TITLE
At uv-export hook. Closes #16.

### LINKED ISSUE #16: Support for `uv export`
It seems sensible to me to support the new `uv export --format requirements-txt` interface as its own hook. For me, this will increasingly become the most common way to generate requirements.txt files rather than the `uv pip compile` interface.

### PR DESCRIPTION
**Pull request by apollo13**

### CODE REVIEW

**Comment by apollo13** [comment]
@zanieb This should be ready for review. I tested it locally and it seems to work okay.

File: README.md [hunk 1]
```diff
@@ -67,6 +67,20 @@ To ensure the lockfile is up-to-date:
...
+```yaml
+- repo: https://github.com/astral-sh/uv-pre-commit
+  # uv version.
+  rev: 0.4.7
+  hooks:
+    # Update the requirements.txt file
+    - id: uv-export
+    # ... or supply a custom file to update:
+    - id: uv-export
+      args: ["--frozen", "--output-file=requirements-custom.txt", "--no-emit-project"]
```
Comment (zanieb) [inline README.md:81]: Should we put this in a separate section like we do for the other examples? It's nice for copy/paste.

File: .pre-commit-hooks.yaml [hunk 2]
```diff
@@ -18,3 +18,13 @@
   pass_filenames: false
   additional_dependencies: []
   minimum_pre_commit_version: "2.9.2"
+- id: uv-export
+  name: uv-export
+  description: "Automatically run 'uv export' on your project dependencies"
+  entry: uv export
+  language: python
+  files: ^uv\.lock$
+  args: ["--frozen", "--output-file=requirements.txt", "--no-emit-project"]
```
Comment (zanieb) [inline .pre-commit-hooks.yaml:27]: Why `--no-emit-project` by default? cc @charliermarsh

**Review by zanieb — COMMENTED** [review]
Thanks for contributing! Some small questions.

File: .pre-commit-hooks.yaml (hunk 2 above)
Comment (apollo13) [reply .pre-commit-hooks.yaml:27]: It feels like the more common usecase to be honest and in line with what other tools do. But that is not a hill I am willing to die on.

Comment (zanieb) [reply .pre-commit-hooks.yaml:27]: I don't think we should differ from the typical defaults without strong justification, it just seems surprising.

Comment (charliermarsh) [reply .pre-commit-hooks.yaml:27]: I'd prefer to omit it. Users can always augment the args in their own pre-commit file, right?

Comment (zanieb) [reply .pre-commit-hooks.yaml:27]: Yeah definitely.

File: README.md [hunk 3]
```diff
@@ -67,6 +67,27 @@ To ensure the lockfile is up-to-date:
...
+
+To supply a custom file for exporting the requirements:
+
+```yaml
+- repo: https://github.com/astral-sh/uv-pre-commit
+  # uv version.
+  rev: 0.4.10
+  hooks:
+    - id: uv-export
+      args: ["--frozen", "--output-file=requirements-custom.txt", "--no-emit-project"]
```
Comment (zanieb) [inline README.md:88]: Note: Drop `--no-emit-project` here too

**Comment by apollo13** [comment]
Thanks for adjusting the PR @zanieb, sorry for not being fast enough :&

What I dislike about having `-e .` (ie the project) in the exported requirements.txt is that it is no longer installable by pip by default (I think this is pip default behavior):
```
ERROR: The editable requirement file:///.../project (from -r requirements.txt (line 2)) cannot be installed when requiring hashes, because there is no single file to hash.
```

So I still feel like that if we really want to export it with the project by default we should export without hashes.

**Comment by zanieb** [comment]
No problem regarding the changes, we often make small touch-ups before merging.

I feel like these are shortcomings of pip, i.e., it enables requiring hashes if it sees a hash and then bans editables despite them not being remote content and there's no way to turn this off. I'd prefer to see these things changed upstream, as they're reasonable behaviors we've implemented in uv. However, I understand that could take a while or never happen so I think the best solution is to add an explicit example to the README explaining how to configure the export if you want to use it with pip.

**Pull request merged**